test/test_question_answering.py

```diff
@@ -140,9 +140,90 @@ def test_id(span_inference_result, no_answer_inference_result):
     assert no_answer_inference_result.id == "best_id_ever"
 
 
+def test_duplicate_answer_filtering(bert_base_squad2):
+    qa_input = [
+        {
+            "questions": ["“In what country lies the Normandy?”"],
+            "text": """The Normans (Norman: Nourmands; French: Normands; Latin: Normanni) were the people who in the 10th and 11th centuries gave their name to Normandy, a region in France. They were descended from Norse (\"Norman\" comes from \"Norseman\") 
+                raiders and pirates from Denmark, Iceland and Norway who, under their leader Rollo, agreed to swear fealty to King Charles III of West Francia. Through generations of assimilation and mixing with the native Frankish and Roman-Gaulish populations, their descendants would gradually merge with the Carolingian-based cultures of West Francia. 
+                The distinct cultural and ethnic identity of the Normans emerged initially in the first half of the 10th century, and it continued to evolve over the succeeding centuries. Weird things happen in Normandy, France."""
+        }]
+
+    bert_base_squad2.model.prediction_heads[0].n_best = 5
+    bert_base_squad2.model.prediction_heads[0].n_best_per_sample = 5
+    bert_base_squad2.model.prediction_heads[0].duplicate_filtering = 0
+
+    result = bert_base_squad2.inference_from_dicts(dicts=qa_input)
+    offset_answer_starts = []
+    offset_answer_ends = []
+    for answer in result[0]["predictions"][0]["answers"]:
+        offset_answer_starts.append(answer["offset_answer_start"])
+        offset_answer_ends.append(answer["offset_answer_end"])
+
+    assert len(offset_answer_starts) == len(set(offset_answer_starts))
+    assert len(offset_answer_ends) == len(set(offset_answer_ends))
+
+
+def test_no_duplicate_answer_filtering(bert_base_squad2):
```

Comment:
I wonder how this test can actually pass, since we already have exact duplicate filtering in place. 
It is used in the PredictionHead when we call formattedpreds to go from Pytorch predictions to actual String Answers.
See https://github.com/deepset-ai/FARM/blob/master/farm/modeling/prediction_head.py#L1386

Reply:
I think here and also in the other tests it would make more sense to list the expected Answers. That way the testing is more explicit, easy to understand and does not open possibilities for unwanted side effects (e.g. tests passing although they should not pass)

Reply:
Interesting. Okay, I will have a closer look and change the tests accordingly.